src/hooks/useUpdateURLQueryParams.ts

```diff
@@ -9,12 +10,15 @@ export const useUpdateURLQueryParams = () => {
   const { search, pathname } = useLocation();
   const updateQueryParams = useCallback(
     (nextQueryParams: StringMap) => {
-      const joinedParams = {
-        ...parseQueryString(search),
-        ...nextQueryParams,
-      };
+      const params = pickBy(
+        {
+          ...parseQueryString(search),
+          ...nextQueryParams,
+        },
+        (v) => v != null
+      );
```

Comment:
Setting a field to undefined will omit it from the query params accomplishing this same effect.